src/GraphQL/Execution/ExecutionStrategy.cs

```diff
@@ -181,16 +181,7 @@ protected virtual async Task ExecuteNodeAsync(ExecutionContext context, Executio
             {
                 resolveContext = new ReadonlyResolveFieldContext(node, context);
 
-                var resolver = node.FieldDefinition.Resolver ?? NameFieldResolver.Instance;
-                var result = resolver.Resolve(resolveContext);
-
-                if (result is Task task)
-                {
-                    await task.ConfigureAwait(false);
-                    result = task.GetResult();
-                }
-
-                node.Result = result;
+                node.Result = await (node.FieldDefinition.Resolver ?? NameFieldResolver.Instance).ResolveAsync(resolveContext);
```

Comment:
Nice for cleanup, but this change will create an object of type Task<object> for every resolved field. I'd advise not to change this section, or permanently change `IFieldResolver.Resolve` to `ResolveAsync`.  See #1527

Reply:
Of note, whether this change is pulled or not, any schema with middleware will be affected by the performance decrease documented in #1380 at the very least

Reply:
Agree about `Task<object>` allocations. I thought about it when I wrote that extension. I decided to postpone the decision until the benchmarks run. Before and after:

|        Method |        Mean |     Error |    StdDev |    Gen 0 |   Gen 1 | Gen 2 | Allocated |
|-------------- |------------:|----------:|----------:|---------:|--------:|------:|----------:|
| Introspection | 2,669.90 us | 22.349 us | 19.812 us | 199.2188 | 54.6875 |     - | 486.73 KB |
|          Hero |    59.27 us |  0.296 us |  0.277 us |   8.8501 |       - |     - |  13.61 KB |


|        Method |        Mean |     Error |     StdDev |      Median |    Gen 0 |   Gen 1 | Gen 2 | Allocated |
|-------------- |------------:|----------:|-----------:|------------:|---------:|--------:|------:|----------:|
| Introspection | 2,899.30 us | 59.505 us | 169.771 us | 2,849.28 us | 214.8438 | 50.7813 |     - | 520.96 KB |
|          Hero |    58.96 us |  0.420 us |   0.328 us |    58.95 us |   8.9111 |       - |     - |  13.73 KB |

I will roll back changes to this file. In two other cases, the use of this method will not get worse, since there have already been `Task<object>` allocations. Agree?

Reply:
Agreed